src/pil/ir/powdr_pil.rs

```diff
@@ -0,0 +1,29 @@
+use crate::{ast::query::Queriable, poly::Expr, util::UUID};
+use std::collections::HashMap;
+extern crate regex;
+
+// PIL circuit IR
+pub struct PilCircuit<F> {
+    // circuit level
+    pub circuit_name: String,
+    pub num_steps: usize,
+
+    // circuit level - col witness
+    pub col_witness: Vec<UUID>, // internal signal NOT dedupped
+
+    // circuit level - col fixed
+    pub fixed_signals: HashMap<UUID, Vec<F>>, // fixed signal UUID -> fixed assignments vector
+    pub step_types_instantiations: HashMap<UUID, Vec<usize>>, /* step type UUID -> step
+                                               * instances
+                                               * vector {0, 1}^num_steps */
+    pub first_step: Option<UUID>,
```

Comment:
@qwang98 same, there is no step in PIL

Reply:
I see your point. I'll try to convert them to constraints in PIL IR.